tools/c7n_mailer/c7n_mailer/slack_delivery.py

```diff
@@ -15,21 +15,20 @@
 
 from botocore.vendored import requests
 import six
-from c7n_mailer.email_delivery import EmailDelivery
 from c7n_mailer.ldap_lookup import Redis
 from c7n_mailer.utils import kms_decrypt, get_rendered_jinja
+from c7n_mailer.utils_email import is_email
 
 
 class SlackDelivery(object):
 
-    def __init__(self, config, session, logger):
-        if config.get('slack_token'):
+    def __init__(self, config, session, logger, email_handler, decrypt=True):
+        if decrypt and config.get('slack_token'):
```

Comment:
I think this is fine, but if you don't want to do this I guess you could also just take the proper token in the constructor (make it responsibility of the caller) and remove `session` and `decrypt`.